lib/statsd/instrument/assertions.rb

```diff
@@ -193,7 +195,7 @@ def assert_statsd_expectations(expectations, datagrams: nil, client: nil, &block
         end
         expectations -= matched_expectations
 
-        unless expectations.empty?
+        unless expectations.select { |m| m.times != 0 }.empty?
```

Comment:
This feels like a double negative with the filter added... how about something like:

```suggestion
        if expectations.any? { |m| m.times != 0 }
```

Reply:
Good point! I integrated your suggestion